File: Pilot.Web/Model/ModifyData/Change.cs
using System.Collections.Generic;
using Ascon.Pilot.DataClasses;

namespace Pilot.Web.Model.ModifyData
{
    public class Change
    {
        public Change()
        {
            Attributes = new ChangesResult<AttributeChangeValue>();
        }

        public string ObjectId { get; set; }
        public ChangesResult<AttributeChangeValue> Attributes { get; set; }
    }

    public class ChangesResult<T>
    {
        public ChangesResult()
        {
            Removed = new List<T>();
            Changed = new List<T>();
        }

        public IList<T> Removed { get; set; }
        public IList<T> Changed { get; set; }
    }

    public class AttributeChangeValue
    {
        public string Name { get; set; }
        public string Value { get; set; }
    }
}

File: Pilot.Web/ClientApp/src/app/core/modifier/change.ts

export class Change {
    constructor(objectId: string) {
        this.attributes = new ChangesResult<AttributeChangeValue>();
        this.objectId = objectId;
    }
    objectId : string;
    attributes : ChangesResult<AttributeChangeValue>;
}

export class ChangesResult<T> {
    constructor() {
        //this.added = new Array<T>()
        this.removed = new Array<T>()
        this.changed = new Array<T>()
    }

    //added: T[];
    removed: T[];
    changed: T[];
}

export class AttributeChangeValue {
    constructor(name: string, value: any) {
        this.name = name;
        this.value = value;
    }
    name: string;
    value: any;
}